### DISCUSSION TITLE
给 ISO 添加 acpid/qemu-ga/open-vm-tools 使其能在PVE/ESXI/UNRAID中正常关机

### DISCUSSION THREAD

**Discussion by ghost** [kkkgo/PaoPaoGateWay#62] [想法]
环境需要 docker 和 git 
1. git clone https://github.com/kkkgo/PaoPaoGateWay.git ppfw 
2. cd ppfw
3. 修改 pkg.conf 在尾部添加 acpid qemu-ga open-vm-tools
4. bash makeiso.sh 等待编译完成
5. cd iso 
6. 此时 iso 目录下已经有一个ISO 如果是DHCP 就可以直接拿来用，如果需要定制则继续
7. docker build . -t ppgw:local
8. 和使用定制iso一样 镜像换成上一步构建的本地镜像 docker run --rm -v .:/data ppgw:local

**Comment by ghost** [comment]
![image](https://github.com/kkkgo/PaoPaoGateWay/assets/93303847/77920465-d7a8-4c8a-ab80-5d9b1245f2c5)
测试已经OK

**Comment by kkkgo** [comment]
已增加acpid，把QEMU Guest Agent禁用应该可以正常关机，可以使用docker定制新镜像。如果需要QEMU Guest Agent，可以使用fullmod定制。

**Reply by kkkgo** [reply]
注：本地编译会开放ssh调试，为了安全建议使用actions的docker定制
